### PR TITLE
Extend PR validation regex to accept dot-separated prerelease identifiers

### PR DESCRIPTION
**Pull request by zcahana**
A recent PR (#464) has made changes to the PR title validation regex so that semantic version strings with [dot-separated prerelease identifiers](https://semver.org/#spec-item-9) are no longer accepted.

Our operator ([redis-enterprise-operator](https://github.com/redhat-openshift-ecosystem/certified-operators/tree/main/operators/redis-enterprise-operator-cert)) is using such dot-separated identifiers as part of its versioning scheme (in recent versions), so can no longer pass validation when submitting new operator versions.

This PR extends the validation regex to also accept dot-separated identifiers in the prerelease section.

Note: the extended regex is still a simplified version, and accepts invalid versions such as `6.4.2-...` or `6.2.4---`.
If favoring accuracy over simplicity, we can switch to use the following regex:
```regex
^operator ([a-zA-Z0-9-]+) \(([0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9](.[a-zA-Z0-9]+)*)?)\)$
```

### CODE REVIEW

**Comment by zcahana** [comment]
cc @jsm84 and @Allda since you were involved in #464.
Appreciate if you guys can take a look at this :-)

**Comment by Allda** [comment]
> cc @jsm84 and @Allda since you were involved in #464. Appreciate if you guys can take a look at this :-)

I fine with the change if @jsm84 confirms this is still a valid version for the operator.

**Comment by jfrancin** [comment]
Looking at the Semantic Versioning Bible (https://www.semver.org), item 9 we see:

> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, 1.0.0-x-y-z.--.

This means that the original PR broke compatibility with Semantic Versioning.  I already have a case on this from Redis Labs (03573527) for just this issue.

**Comment by jsm84** [comment]
@zcahana I'm ready to approve as soon as the linting issues and those mentioned in the review are fixed.

**Comment by zcahana** [comment]
@jsm84 I've fixed the lint issue - specifically this was the black formatter that broke some of the test cases across several lines but kept others on a single line. I figured it would be visually nicer to have all test cases formatted consistently, so I locally disabled the formatting for the test cases.

**Comment by zcahana** [comment]
> I already have a case on this from Redis Labs (03573527) for just this issue

@jfrancin that's quite of a coincidence, as I'm from Redis (ex-Labs) too! ;-)

**Review by jsm84 — APPROVED** [review]

File: operator-pipeline-images/tests/test_operatorcert.py [hunk 1]
```diff
@@ -269,13 +269,26 @@ def test_verify_changed_files_location(wrong_change: str):
...
+            True,
+            "operator-test123",
+            "1.0.1-8.0",
+        ),
+        (
+            "operator operator-test123 (1.0.1-rc.1.0)",
+            True,
+            "operator-test123",
+            "1.0.1-rc.1.0",
+        ),
```
[the remark this answers is not in the document]
Comment (jsm84) [reply operator-pipeline-images/tests/test_operatorcert.py]: looks good, thanks!

**Pull request merged**